src/plugins/logger/loggerplugin.cpp

```diff
@@ -0,0 +1,105 @@
+      from this software without specific prior written permission.
+
+  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+  ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+  WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+  DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDERS OR CONTRIBUTORS BE LIABLE FOR
+  ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+  (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+  ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+  (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*/
+
+#include "loggerplugin.h"
+#include <IrcConnection>
+#include <IrcNetwork>
+#include <IrcMessage>
+#include <IrcCommand>
+#include <IrcChannel>
+#include <Irc>
+#include <QDir>
+#include <QTextStream>
+
+LoggerPlugin::LoggerPlugin(QObject* parent) : QObject(parent)
+    , m_logDirPath(QDir::homePath()+"/communilogs/")
+{
+    QDir logDir;
+    if (!logDir.exists(m_logDirPath))
+        logDir.mkpath(m_logDirPath);
+}
+
+LoggerPlugin::~LoggerPlugin()
+{
+}
+
+void LoggerPlugin::bufferAdded(IrcBuffer* buffer)
+{
+    // Do not log connection buffers and #magna
+    if (buffer->network()->name().isEmpty())
+        return;
+
+    const QString filename = logfileName(buffer);
+    writeToFile(filename, "=== Logfile started on " + timestamp() + " ===");
+
+    connect(buffer, SIGNAL(messageReceived(IrcMessage*)), this, SLOT(logMessage(IrcMessage*)));
+}
+
+void LoggerPlugin::bufferRemoved(IrcBuffer* buffer)
+{
+    disconnect(buffer, SIGNAL(messageReceived(IrcMessage*)), this, SLOT(logMessage(IrcMessage*)));
+}
+
+void LoggerPlugin::logMessage(IrcMessage *message)
+{
+    if (message->type() != IrcMessage::Private)
+        return;
+
+    IrcPrivateMessage *m = static_cast<IrcPrivateMessage*>(message);
+    const QString filename = logfileName(m);
+    writeToFile(filename , timestamp() + " " + m->nick() + ": " + m->content());
+}
+
+void LoggerPlugin::writeToFile(const QString &fileName, const QString &text)
+{
+    QFile logfile(m_logDirPath + fileName);
+    logfile.open(QIODevice::WriteOnly | QIODevice::Append | QIODevice::Text);
+
+    QTextStream ts(&logfile);
+    ts << text << endl;
+
+    ts.setDevice(0);
+    logfile.close();
+}
+
+QString LoggerPlugin::logfileName(IrcPrivateMessage *message) const
+{
+    return message->isPrivate() ? message->network()->name() + "_" + message->nick() + ".log"
+                                : message->network()->name() + "_" + message->target() + ".log";
+}
+
+QString LoggerPlugin::logfileName(IrcBuffer *buffer) const
+{
+    return buffer->isChannel() ? buffer->network()->name() + "_#" + buffer->name() + ".log"
```

Comment:
i suppose this would do the job too:

    return buffer->network()->name() + buffer->title() + ".log";